cmd/thanos/compact.go

```diff
@@ -163,8 +174,32 @@ func runCompact(
 						select {}
 					}
 				}
-				return nil
-			})
+				if !didWork {
+					break
+				}
+			}
+			// After all comapctions are done, work down the downsampling backlog.
+			// We run two passes of this to ensure that the 1h downsampling is generated
+			// for 5m downsamplings created in the first run.
+			level.Info(logger).Log("msg", "start first pass of downsampling")
+
+			if err := downsampleBucket(ctx, logger, bkt, dataDir); err != nil {
+				return errors.Wrap(err, "downsampling failed")
```

Comment:
worth to make this wrap msg unique? 

`s/downsampling failed/first phase downsampling failed`

Reply:
not addressed